src/types/server_decoration.zig

```diff
@@ -0,0 +1,52 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+pub const ServerDecorationManager = extern struct {
+    pub const Mode = enum(c_int) {
+        none = 0,
+        client = 1,
+        server = 2,
+    };
+
+    global: *wl.Global,
+    resources: wl.list.Head(wl.Resource, null),
+    decorations: wl.list.Head(ServerDecoration, "link"),
+
+    default_mode: Mode,
+
+    server_destroy: wl.Listener(*wl.Server),
+
+    events: extern struct {
+        new_decoration: wl.Signal(*ServerDecoration),
+        destroy: wl.Signal(*ServerDecorationManager),
+    },
+
+    data: usize,
+
+    extern fn wlr_server_decoration_manager_create(server: *wl.Server) ?*ServerDecorationManager;
+    pub fn create(server: *wl.Server) !*ServerDecorationManager {
+        return wlr_server_decoration_manager_create(server) orelse error.OutOfMemory;
+    }
+
+    extern fn wlr_server_decoration_manager_set_default_mode(manager: *ServerDecorationManager, default_mode: Mode) void;
```

Comment:
again, `uint32_t` does not have the same ABI as an enum on all targets. Feel free to accept the enum in a zig wrapper function and cast though.

Alternatively we could make the enum's backing integer a `u32`.